model.js

```diff
@@ -4,4 +4,36 @@ const utils = require("./db/seeds/utils")
 exports.selectTopics = () => {
     return db.query("SELECT * FROM topics")
     .then(data => data.rows)
+}
+
+exports.selectArticleById = (article_id) => {
+    
+    return db.query(`SELECT * FROM articles WHERE article_id=$1`, [article_id])
+    .then(({rows}) => {
+        if (rows.length ===0){
+            return Promise.reject({status: 404, message: "no article_id found"})
+        } else {
+            console.log(rows[0], article_id)
+            return db.query(`SELECT * FROM comments WHERE article_id=$1`, [article_id])
+            .then(data => {
+                rows[0].comment_count=data.rows.length
+                return rows[0]
```

Comment:
This could/should be done with one query to the database where you use a 'group by' and 'aggregate' function.

Reply:
I will need to come back to this later